Sources/SwiftParser/Specifiers.swift

```diff
@@ -520,6 +633,65 @@ extension Parser {
   mutating func parseAccessorEffectSpecifiers() -> RawAccessorEffectSpecifiersSyntax? {
     return parseEffectSpecifiers(RawAccessorEffectSpecifiersSyntax.self)
   }
+    
+  mutating func parseDeinitEffectSpecifiers() -> RawDeinitEffectSpecifiersSyntax? {
+    // Note that parseEffectSpecifiers() consumes deinit name as unexpected token
+    // But we want it to be handled on the higher level.
+    // So we parseEffectSpecifiers() is not reused here.
+    
+    var unexpectedBeforeAsync: [RawSyntax?] = []
+    var asyncKeyword: RawTokenSyntax?
+    var unexpectedAfterAsync: [RawSyntax?] = []
+    
+    while true {
```

Comment:
This also seems overly complicated. Can’t we just eat consume any misplaced throws specifiers into `unexpectedBeforeAsync` and then just do `self.consume(if: .async)`?

Alternatively, could we keep your hand-written initializer for `RawDeinitEffectSpecifiersSyntax`, make it conform to `RawEffectSpecifiersTrait` and use `parseEffectSpecifiers`?

Reply:
I was trying to follow the logic of `parseEffectSpecifiers()` and handle the same recovery cases, including misspelled and duplicated `async`. If I understood you correctly, this simplification would not parse such cases, or would it?

I've tried the latter, but it consumes deinit name into `unexpectedBeforeAsyncSpecifier`.

Reply:
I would then combine all the loops into one.

Reply:
I've pushed an update, please take a look